src/realm/array_blob.cpp

```diff
@@ -142,7 +142,7 @@ ref_type ArrayBlob::replace(size_t begin, size_t end, const char* data, size_t d
         return new_root.blob_replace(begin, end, data, data_size, add_zero_term);
     }
 
-    if (remove_size == add_size && is_read_only() && memcmp(m_data + begin, data, data_size) == 0)
+    if (remove_size == add_size && is_read_only() && (data_size == 0 || memcmp(m_data + begin, data, data_size) == 0))
```

Comment:
What was this one about? Does `memcmp` actually access the memory if the length is 0?
I'm trying to understand if this could be a fix for stack traces like the one reported in https://github.com/realm/realm-core/issues/6463

Reply:
memcmp is defined as taking non-null parameters and passing null is UB. In practice all implementations are fine with null if the length is zero but UBSan complains anyway.

Reply:
Thanks, good to know.